pynest/examples/pynest-microcircuit/network.py

```diff
@@ -0,0 +1,347 @@
+# -*- coding: utf-8 -*-
+#
+# network.py
+#
+# This file is part of NEST.
+#
+# Copyright (C) 2004 The NEST Initiative
+#
+# NEST is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# NEST is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+
+'''
+pynest microcicuit network
+----------------------------------------------
+
+Main file for the microcircuit.
+
+Hendrik Rothe, Hannah Bos, Sacha van Albada; May 2016
+'''
+
+import nest
+import numpy as np
+from helpers import *
+import matplotlib.pyplot as plt
+import os
+
+
+class Network:
+    """ Handles the setup of the network parameters and
+    provides functions to connect the network and devices.
+
+    Arguments:
+    sim_dict: dictionary containing all parameters specific to the simulation
+              such as the directory the data is stored in and the seeds
+              (see: stimulus_params.py)
+    net_dict: dictionary containing all parameters specific to the neurons
+              and the network
+              (see: network_params.py)
+
+    Keyword Arguments:
+    stim_dict: dictionary containing all parameter specific to the stimulus
+               (see: stimulus_params.py)
+    """
+    def __init__(self, sim_dict, net_dict, stim_dict=None):
+        self.sim_dict = sim_dict
+        self.data = sim_dict['data_path']
+        self.net_dict = net_dict
+        if stim_dict is not None:
+            self.stim_dict = stim_dict
+        print 'Data will be written to %s' % self.data
+
```

Comment:
All members of the class should be initialized in the `__init__` function and not in the member functions.